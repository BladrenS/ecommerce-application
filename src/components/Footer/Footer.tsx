import { type FC, memo } from 'react';
import { NavLink, useLocation } from 'react-router-dom';

import { location, logo, mail, phone } from '../../assets/';
import { footerLinks, paymentIcons, promoItems, socialIcons } from '../../constants/';
import styles from './styles.module.scss';

export const Footer: FC = memo(() => {
  const { pathname } = useLocation();
  const handleClick = (id: string | undefined) => {
    if (id) {
      localStorage.setItem('filter', id);
    }
  };
  if (pathname === '/catalog') {
    for (const item of footerLinks[2].links) {
      item.disabled = true;
    }
  } else {
    for (const item of footerLinks[2].links) {
      item.disabled = false;
    }
  }

  return (
    <footer className={styles.footer}>
      <div className={styles['footer-container']}>
        <div className={styles['footer-promo']}>
          {promoItems.map((item, i) => (
            <div key={i} className={styles['footer-promo-item']}>
              <img src={item.img} alt={`promo-${i}`} />
              <span className={styles.span}>{item.text}</span>
            </div>
          ))}
          <div className={styles['footer-promo-item']}>
            <h3>Want to be the first to know about all our deals? Subscribe now!</h3>
            <div className={styles['input-block']}>
              <input
                name="email-field"
                type="text"
                placeholder="enter your email address..."
                className={styles.input}
              />
              <div className={styles['yes-button']}>Yes</div>
            </div>
            <span className={styles.span}>
              Subscribe to our newsletter for updates on new arrivals, exclusive discounts, and notifications about our
              latest blog articles. Stay informed and enhance your moto experience with our expert tips and offers!
            </span>
          </div>
        </div>
      </div>

      <div className={styles['footer-connections']}>
        <NavLink to="/main" className={styles.shop}>
          <img src={logo} alt="logo" />
          хрустикshop
        </NavLink>
        <a
          className={styles['footer-connections-item']}
          href="https://maps.app.goo.gl/1jSjmqpZzkUWpoZJ8"
          target="_blank"
        >
          <img src={location} alt="mark" />
          Николаева ул., 2/1, Москва, 103274
        </a>
        <a href="mailto:connect@хрустикshop.com" className={styles['footer-connections-item']}>
          <img src={mail} alt="mail" />
          connect@хрустикshop.com
        </a>
        <a href="tel:+78005552525" className={styles['footer-connections-item']}>
          <img src={phone} alt="phone" />
          +7 800 555 25 25
        </a>
      </div>

      <div className={styles['footer-container']}>
        {footerLinks.map((block, index) => (
          <div className={styles['footer-links-block']} key={index}>
            <h2 className={styles['links-header']}>{block.header}</h2>
            {block.links.map((link, i) =>
              link.disabled ? (
                <div key={i} className={styles.disabled}>
                  {link.label}
                </div>
              ) : (
                <NavLink key={i} to={link.to} className={styles['footer-link']} onClick={() => handleClick(link.id)}>
                  {link.label}
                </NavLink>
              ),
            )}
          </div>
        ))}

        <div className={styles['footer-icons-block']}>
          <h2 className={styles.paragraph}>Social Media</h2>
          <div className={styles['icons-container']}>
            {socialIcons.map((icon, i) => (
              <img key={i} src={icon} alt={`social-${i}`} className={styles.icon} />
            ))}
          </div>
          <h2 className={styles.paragraph}>We accept</h2>
          <div className={styles['icons-container']}>
            {paymentIcons.map((icon, i) => (
              <img key={i} src={icon} alt={`payment-${i}`} width="32" />
            ))}
          </div>
        </div>
      </div>
    </footer>
  );
});
