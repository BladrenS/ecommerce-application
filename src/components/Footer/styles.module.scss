@use '../../styles/abstract/mixins' as *;

.footer {
  background-color: var(--background-color-footer);
}

.footer-container {
  display: flex;
  flex-wrap: wrap;
  gap: 30px;
  align-items: center;
  justify-content: space-between;

  margin: 0 auto;
  padding: 0 40px;

  @media (width <= 478px) {
    justify-content: space-around;
  }
}

.footer-promo {
  display: grid;
  grid-template-columns: repeat(4, 1fr);

  @media (width <= 1200px) {
    grid-template-columns: repeat(2, 1fr);
  }

  @media (width <= 768px) {
    grid-template-columns: repeat(1, 1fr);
  }
}

.footer-promo-item {
  flex-direction: column;

  margin: 26px 0;
  padding: 20px;
  border-right: 1px solid #ff620025;

  color: var(--primary-color);

  @include flex-center;

  &:last-child {
    border-right: none;
  }
}

.span {
  margin-top: 40px;
}

.input-block {
  width: 100%;
  margin-top: 30px;

  @include flex-center;
}

.input {
  width: 70%;
  height: 50px;
  padding-left: 10px;
  border: none;
  border-top-left-radius: 8px;
  border-bottom-left-radius: 8px;

  &:focus {
    outline: none;
  }
}

.yes-button {
  width: 70px;
  height: 50px;
  border-top-right-radius: 8px;
  border-bottom-right-radius: 8px;

  font-weight: bold;
  color: var(--secondary-color);

  background-color: var(--primary-color);

  transition: all 0.2s linear;

  @include flex-center;

  &:hover {
    cursor: pointer;
    background-color: var(--primary-hover);
  }
}

.shop {
  display: flex;
  gap: 15px;
  align-items: center;

  font-size: 26px;
  font-weight: bold;
  color: var(--primary-color);
  text-transform: uppercase;
}

.footer-connections {
  display: flex;
  flex-wrap: wrap;
  gap: 30px;
  justify-content: space-between;

  padding: 20px 40px;

  background-color: rgb(26 25 25);

  @media (width <= 400px) {
    padding: 20px 5px;
  }
}

.footer-connections-item {
  display: flex;
  gap: 15px;
  align-items: center;
  justify-content: center;

  color: var(--primary-color);

  opacity: 0.7;

  transition: all 0.2s linear;

  &:hover {
    opacity: 1;
  }
}

.footer-links-block {
  display: flex;
  flex-direction: column;
  margin: 26px 0;
  color: var(--secondary-color);
}

.links-header {
  margin-bottom: 30px;
}

.footer-link {
  margin: 15px 0;
  transition: all 0.2s linear;

  &:hover {
    color: var(--primary-color);
  }
}

.disabled {
  margin: 15px 0;
  opacity: 0.3;

  &:hover {
    cursor: not-allowed;
  }
}

.footer-icons-block {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;

  height: 220px;
  margin: 26px 0;
}

.icon {
  width: 40px;
  padding: 5px;
  border: 1px solid var(--primary-color);
  border-radius: 5px;

  &:hover {
    cursor: pointer;
    background-color: #ff62002c;
  }
}

.paragraph {
  color: var(--secondary-color);
}

.icons-container {
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 230px;
}
