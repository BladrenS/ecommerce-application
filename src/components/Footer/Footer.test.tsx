import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import { Footer } from './Footer';

describe('Footer snapshot test', () => {
  it('renders correctly', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Footer />
      </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
