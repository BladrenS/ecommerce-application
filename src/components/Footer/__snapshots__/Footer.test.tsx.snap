// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer snapshot test renders correctly 1`] = `
<DocumentFragment>
  <footer
    class="footer"
  >
    <div
      class="footer-container"
    >
      <div
        class="footer-promo"
      >
        <div
          class="footer-promo-item"
        >
          <img
            alt="promo-0"
            src="test-file-stub"
          />
          <span
            class="span"
          >
            Gear up for the ride! This week only – special discounts on all motorcycle helmets. Stay safe and stylish without breaking the bank!
          </span>
        </div>
        <div
          class="footer-promo-item"
        >
          <img
            alt="promo-1"
            src="test-file-stub"
          />
          <span
            class="span"
          >
            Rev up your savings! Get top-quality motorcycle parts at unbeatable prices — for a limited time only. Keep your ride in peak condition without draining your wallet!
          </span>
        </div>
        <div
          class="footer-promo-item"
        >
          <img
            alt="promo-2"
            src="test-file-stub"
          />
          <span
            class="span"
          >
            Chain up and ride strong! Premium motorcycle chains now on sale — built for durability, priced for riders. Don’t miss the deal that keeps you moving!
          </span>
        </div>
        <div
          class="footer-promo-item"
        >
          <h3>
            Want to be the first to know about all our deals? Subscribe now!
          </h3>
          <div
            class="input-block"
          >
            <input
              class="input"
              name="email-field"
              placeholder="enter your email address..."
              type="text"
            />
            <div
              class="yes-button"
            >
              Yes
            </div>
          </div>
          <span
            class="span"
          >
            Subscribe to our newsletter for updates on new arrivals, exclusive discounts, and notifications about our latest blog articles. Stay informed and enhance your moto experience with our expert tips and offers!
          </span>
        </div>
      </div>
    </div>
    <div
      class="footer-connections"
    >
      <a
        class="shop"
        data-discover="true"
        href="/main"
      >
        <img
          alt="logo"
          src="test-file-stub"
        />
        хрустикshop
      </a>
      <a
        class="footer-connections-item"
        href="https://maps.app.goo.gl/1jSjmqpZzkUWpoZJ8"
        target="_blank"
      >
        <img
          alt="mark"
          src="test-file-stub"
        />
        Николаева ул., 2/1, Москва, 103274
      </a>
      <a
        class="footer-connections-item"
        href="mailto:connect@хрустикshop.com"
      >
        <img
          alt="mail"
          src="test-file-stub"
        />
        connect@хрустикshop.com
      </a>
      <a
        class="footer-connections-item"
        href="tel:+78005552525"
      >
        <img
          alt="phone"
          src="test-file-stub"
        />
        +7 800 555 25 25
      </a>
    </div>
    <div
      class="footer-container"
    >
      <div
        class="footer-links-block"
      >
        <h2
          class="links-header"
        >
          General
        </h2>
        <a
          class="footer-link"
          data-discover="true"
          href="/profile"
        >
          My account
        </a>
        <a
          class="footer-link"
          data-discover="true"
          href="/about"
        >
          About us
        </a>
        <a
          class="footer-link"
          data-discover="true"
          href="/blog"
        >
          Blog
        </a>
        <a
          aria-current="page"
          class="disabled active"
          data-discover="true"
          href="/"
        >
          Career
        </a>
        <a
          class="footer-link"
          data-discover="true"
          href="/Cooperation"
        >
          Cooperation
        </a>
      </div>
      <div
        class="footer-links-block"
      >
        <h2
          class="links-header"
        >
          Help & Guide
        </h2>
        <a
          aria-current="page"
          class="disabled active"
          data-discover="true"
          href="/"
        >
          Help center
        </a>
        <a
          aria-current="page"
          class="disabled active"
          data-discover="true"
          href="/"
        >
          How to buy
        </a>
        <a
          aria-current="page"
          class="disabled active"
          data-discover="true"
          href="/"
        >
          Delivery
        </a>
        <a
          aria-current="page"
          class="disabled active"
          data-discover="true"
          href="/"
        >
          Product Policy
        </a>
        <a
          aria-current="page"
          class="disabled active"
          data-discover="true"
          href="/"
        >
          How to Return
        </a>
      </div>
      <div
        class="footer-links-block"
      >
        <h2
          class="links-header"
        >
          Categories
        </h2>
        <a
          class="footer-link"
          data-discover="true"
          href="/profile"
        >
          Cat
        </a>
        <a
          class="footer-link"
          data-discover="true"
          href="/about"
        >
          Cat
        </a>
        <a
          class="footer-link"
          data-discover="true"
          href="/blog"
        >
          Cat
        </a>
        <a
          aria-current="page"
          class="footer-link active"
          data-discover="true"
          href="/"
        >
          Cat
        </a>
        <a
          class="footer-link"
          data-discover="true"
          href="/Cooperation"
        >
          Cat
        </a>
      </div>
      <div
        class="footer-icons-block"
      >
        <h2
          class="paragraph"
        >
          Social Media
        </h2>
        <div
          class="icons-container"
        >
          <img
            alt="social-0"
            class="icon"
            src="test-file-stub"
          />
          <img
            alt="social-1"
            class="icon"
            src="test-file-stub"
          />
          <img
            alt="social-2"
            class="icon"
            src="test-file-stub"
          />
          <img
            alt="social-3"
            class="icon"
            src="test-file-stub"
          />
          <img
            alt="social-4"
            class="icon"
            src="test-file-stub"
          />
        </div>
        <h2
          class="paragraph"
        >
          We accept
        </h2>
        <div
          class="icons-container"
        >
          <img
            alt="payment-0"
            src="test-file-stub"
            width="32"
          />
          <img
            alt="payment-1"
            src="test-file-stub"
            width="32"
          />
          <img
            alt="payment-2"
            src="test-file-stub"
            width="32"
          />
          <img
            alt="payment-3"
            src="test-file-stub"
            width="32"
          />
        </div>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
